pkg/controllers/cluster/cluster_controller.go

```diff
@@ -511,3 +533,37 @@ func (c *Controller) processTaintBaseEviction(ctx context.Context, cluster *clus
 	}
 	return nil
 }
+
+func (c *Controller) taintClusterByCondition(ctx context.Context, cluster *clusterv1alpha1.Cluster) error {
+	// Cluster object may be changed in this function.
+	observedReadyCondition, currentReadyCondition, err := c.tryUpdateClusterHealth(ctx, cluster)
```

Comment:
There is no need to update cluster health when reconciling because it is a little bit expensive. I suggest retrieve conditions from cluster status directly. BTW, [here](https://github.com/kubernetes/kubernetes/blob/1868a3a2eb9945f26239f700fc491fb37232db5a/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L611-L665) is some references.

Reply:
done